attest/verifier/build.rs

```diff
@@ -72,213 +66,10 @@ impl RngCallback for RngForMbedTls {
         null_mut()
     }
 }
-fn purge_expired_cert(path: &Path) {
-    let mut bytes = match read(path) {
-        Ok(bytes) => bytes,
-        Err(_) => {
-            return;
-        }
-    };
-
-    // mbedtls says "Input must be NULL-terminated".
-    bytes.push(0);
-
-    match Certificate::from_pem_multiple(&bytes).map(|certs| {
-        let ts = certs
-            .iter()
-            .last()
-            .expect("no certs")
-            .not_after()
-            .unwrap_or_else(|e| panic!("invalid certificate expiration time: {e:?}"));
-        Utc.with_ymd_and_hms(
-            ts.year as i32,
-            ts.month as u32,
-            ts.day as u32,
-            ts.hour as u32,
-            ts.minute as u32,
-            ts.second as u32,
-        )
-        .single()
-        .unwrap_or_else(|| panic!("Invalid expiration time: {ts:?}."))
-    }) {
-        Ok(not_after) => {
-            let utc_now = Utc::now();
-
-            // If certificate expired or expires in the next 24 hours, delete it so it gets
-            // regenerated.
-            if utc_now + Duration::hours(24) > not_after {
-                remove_file(path)
-                    .unwrap_or_else(|e| panic!("failed deleting expired cert {path:?}: {e:?}"));
-            }
-        }
-        Err(_) => {
-            // Failed getting expiration date from certificate, delete it so it gets
-            // regenerated.
-            remove_file(path)
-                .unwrap_or_else(|e| panic!("failed deleting non-parseable cert {path:?}: {e:?}"));
-        }
-    }
-}
 
 fn main() {
-    // Generate simulation IAS certificates
     let base_dir = env::var("CARGO_MANIFEST_DIR").expect("Could not read manifest dir");
-    let mut data_path = PathBuf::from(base_dir);
-    data_path.push("data");
-    data_path.push("sim");
-
-    let mut root_anchor_path = data_path.clone();
-    root_anchor_path.push("root_anchor.pem");
-
-    let mut signer_key_path = data_path.clone();
-    signer_key_path.push("signer.key");
-
-    let mut chain_path = data_path;
-    chain_path.push("chain.pem");
-
-    cargo_emit::rerun_if_env_changed!("MC_SEED");
-    cargo_emit::rerun_if_changed!(root_anchor_path
-        .to_str()
-        .expect("Could not stringify root anchor path"));
-    cargo_emit::rerun_if_changed!(signer_key_path
-        .to_str()
-        .expect("Could not stringify signer key path"));
-    cargo_emit::rerun_if_changed!(chain_path
-        .to_str()
-        .expect("Could not stringify signer chain path"));
-
-    purge_expired_cert(&root_anchor_path);
-    purge_expired_cert(&chain_path);
-
-    if !(root_anchor_path.exists() && signer_key_path.exists() && chain_path.exists())
-        || env::var("MC_SEED").is_ok()
```

Comment:
I may misunderstanding what the intent here was.

My understanding is that Cargo will not invoke this build.rs file unless one of the "rerun_if_changed" files has been touched, or the MC_SEED variable has changed.

This means that the "purge_expired_cert" won't normally be hit.

Based on the expiry time below it seems like one would need to somehow manually kick this every week, I know the comment says 30 days, but the code is one week.

```rust
        // Good for 30 days
        let end_time = end_now + Duration::weeks(1);
```
I haven't been playing with enclave verification specifically to know what people have been doing because I'm pretty sure that after a week a manual intervention of some kind is necessary.

Reply:
That might've just been an oversight. I will note that I have seen the `mc-attest-verifier` crate being rebuilt way more often than necessary in multiple occasions, so perhaps that masked this issue. I never dug into what was causing it (and it's possible it has since been resolved - it's been awhile since I spent too much time building this repo to notice the build behavior)